src/dune_rules/melange/melange_rules.ml

```diff
@@ -16,36 +16,39 @@ let ocaml_flags sctx ~dir melange =
     let ocaml_version = (Super_context.context sctx).version in
     Super_context.with_vendored_flags ~ocaml_version flags
 
-let lib_output_dir ~sctx ~target_dir lib =
+let output_of_lib ~target_dir lib =
   let info = Lib.info lib in
-  let lib_dir =
-    match Lib_info.status info with
-    | Private _ ->
-      let lib = Lib.Local.of_lib_exn lib in
-      let info = Lib.Local.info lib in
-      Lib_info.src_dir info
-    | Public _ ->
-      let package_name = Option.value_exn (Lib_info.package info) in
-      let bctx = (Super_context.context sctx).build_dir in
```

Comment:
was this previously wrong or just redundant?

Reply:
It was definitely wrong as it was using the `src_dir` of public libraries, but I think I kept that behavior. The only thing I changed is that there's no need to worry about library directories at all for private libraries since we can derive the .js path directly from the source path.

Reply:
I was referring to `(Super_context.context sctx).build_dir`, which now I forget what it does.

Reply:
Oh, yeah that's definitely wrong. Because that's outputting packages outside target directory. `bctx` would be a path like `_build/default`